### PR TITLE
feat(indy): add choice for taa mechanism

### PR DESCRIPTION
**Pull request by morrieinmaas**
todos:
* tests
* ensure this is complete and correct

Signed-off-by: Moriarty <moritz@animo.id>

BREAKING CHANGE: the transaction author agreement acceptance mechanism was previously automatically the first acceptance mechanism from the acceptance mechanism list. With this addition, the framework never automatically selects the acceptance mechanism anymore and it needs to be specified in the `transactionAuthorAgreement` in the `indyLedgers` agent config array.

### CODE REVIEW

**Comment by morrieinmaas** [comment]
closes #707

File: packages/core/src/modules/ledger/IndyPool.ts [hunk 1]
```diff
@@ -9,11 +9,17 @@ import { isIndyError } from '../../utils/indyError'
 import { LedgerError } from './error/LedgerError'
 import { isLedgerRejectResponse, isLedgerReqnackResponse } from './ledgerUtil'
 
+export interface TransactionAuthorAgreement {
+  version: `${number}.${number}` | `${number}`
```
Comment (morrieinmaas) [inline packages/core/src/modules/ledger/IndyPool.ts:13]: suggestion in issue description was ``version: `${number}.${number}` `` only. Maybe I'm missing sth but version I found was of format `` `${number}` `` so I added that as alternative type/format

**Comment by morrieinmaas** [comment]
@TimoGlastra @blu3beri @karimStekelenburg any feedback welcome

File: packages/core/src/modules/ledger/LedgerModule.ts [hunk 2]
```diff
@@ -26,47 +27,58 @@ export class LedgerModule {
     await this.ledgerService.connectToPools()
   }
 
-  public async registerPublicDid(did: string, verkey: string, alias: string, role?: NymRole) {
```
Comment (TimoGlastra) [inline packages/core/src/modules/ledger/LedgerModule.ts:29]: If we add the TAA to the indy pool config, there's no need to specify it for each method we call. We can extract it from the indy pool config at a lower level

File: packages/core/src/modules/ledger/services/IndyLedgerService.ts [hunk 3]
```diff
@@ -56,7 +56,8 @@ export class IndyLedgerService {
     targetDid: string,
     verkey: string,
     alias: string,
-    role?: NymRole
+    role?: NymRole,
+    taa?: TransactionAuthorAgreement
```
Comment (TimoGlastra) [inline packages/core/src/modules/ledger/services/IndyLedgerService.ts:60]: same here, we can extract it on a lower level

File: packages/core/src/modules/ledger/services/IndyLedgerService.ts [hunk 4]
```diff
@@ -451,21 +458,43 @@ export class IndyLedgerService {
...
+        throw new Error(errMessage)
+      }
 
       const requestWithTaa = await this.indy.appendTxnAuthorAgreementAcceptanceToRequest(
         request,
         authorAgreement.text,
-        authorAgreement.version,
+        taa.version,
         authorAgreement.digest,
-        this.getFirstAcceptanceMechanism(authorAgreement),
```
Comment (TimoGlastra) [inline packages/core/src/modules/ledger/services/IndyLedgerService.ts:468]: we can delete this method now

File: packages/core/src/modules/ledger/services/IndyLedgerService.ts [hunk 5]
```diff
@@ -451,21 +458,43 @@ export class IndyLedgerService {
     }
   }
 
-  private async appendTaa(pool: IndyPool, request: Indy.LedgerRequest) {
+  private async appendTaa(pool: IndyPool, request: Indy.LedgerRequest, taa?: TransactionAuthorAgreement) {
```
Comment (TimoGlastra) [inline packages/core/src/modules/ledger/services/IndyLedgerService.ts:461]: We pass the pool config to `IndyPool` in the constructor. If we make the taa config available on the `InydPool` we don't have to pass this value around everywhere

**Review by TimoGlastra — CHANGES_REQUESTED** [review]

**Comment by TimoGlastra** [comment]
Also could you add some tests?

**Review by TimoGlastra — APPROVED** [review]
This is great @morrieinmaas! 

The new `transactionAuthorAgreement` needs to be documented. Could you also create a PR in aries-javascript-docs to add this property? It should be added here: https://github.com/animo/aries-javascript-docs/blob/main/guides/tutorials/agent-config.md#indyledgers

**Comment by morrieinmaas** [comment]
> This is great @morrieinmaas!
> 
> The new `transactionAuthorAgreement` needs to be documented. Could you also create a PR in aries-javascript-docs to add this property? It should be added here: https://github.com/animo/aries-javascript-docs/blob/main/guides/tutorials/agent-config.md#indyledgers

Yes I will do!

**Comment by 2byrds** [comment]
The documentation is growing nicely!

**Pull request merged**